app/controllers/registration_controller.rb

```diff
@@ -286,50 +282,43 @@ def validate_guests!
     end
 
     def guests_valid?
-      @competition[:competition_info]["guest_entry_status"] != "restricted" || @competition[:competition_info]["guests_per_registration_limit"] >= params[:guests]
+      @competition.guest_entry_status != 'restricted' || @competition.guest_limit >= params[:guests]
```

Comment:
I think it makes sense to have this as a guest_allowed method in the competition classe

Reply:
Good point! Have changed it to `guest_limit_exceeded`, as that flowed better with the logic of checking for the `:guests` key first